src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreBuildHandler.cs

```diff
@@ -141,8 +144,9 @@ public async Task<bool> RestoreAsync(uint buildAction, CancellationToken token)
             if ((buildAction & (uint)VSSOLNBUILDUPDATEFLAGS.SBF_OPERATION_CLEAN) != 0 &&
                 (buildAction & (uint)VSSOLNBUILDUPDATEFLAGS3.SBF_FLAGS_UPTODATE_CHECK) == 0)
             {
-                // Clear the project.json restore cache on clean to ensure that the next build restores again
+                // Clear the transitive restore cache on clean to ensure that the next build restores again
                 await SolutionRestoreWorker.Value.CleanCacheAsync();
+                SolutionRestoreChecker.Value.CleanCache();
```

Comment:
Is this PR not complete yet? I don't see any actual logic for how this SolutionRestoreChecker is helping to make fast the process.

Reply:
I'm not sure I follow the concern. 

The clean cache part is about backwards compat with the gestures. Given that there are 2 no-ops, we want to ensure a cache delete does affect the up to date check. 

So whenever a clean gesture occurs, we ensure to tell the SolutionRestoreChecker about it.